python-sdk/pachyderm_sdk/api/storage/extension.py

```diff
@@ -0,0 +1,166 @@
+    CdrResolver,
+    Ref,
+    Cipher,
+    Compress,
+    Concat,
+    ContentHash,
+    Http,
+    SizeLimits,
+    Slice,
+)
+from . import FilesetStub as _GeneratedApiStub
+from . import FileFilter, PathRange
+from pachyderm_sdk.constants import CDR_CACHE_LOCATION
+
+
+class ApiStub(_GeneratedApiStub):
+
+    def assemble_fileset(
+        self,
+        fileset: str,
+        path: Union[str, PathRange],
+        destination: os.PathLike,
+        *,
+        cache_location: os.PathLike = CDR_CACHE_LOCATION,
+        fetch_missing_chunks: bool = True,
+        http_host_replacement: str = "",
+    ) -> None:
+        """Assemble the data contained at the specified path within the fileset
+        and write that data to the specified destination directory. This method
+        uses a local cache and will automatically fetch missing data from PFS,
+        unless disabled.
+
+        Parameters
+        ----------
+        fileset : str
+            The UUID of the fileset.
+        path : str
+            The path, regex, or PathRange within the fileset to assemble.
+            A value of "/" assembles the entire fileset.
+        destination : os.PathLike
+            The local directory at which to assemble the fileset.
+        cache_location : os.PathLike (kwarg only)
+            The location of the chunk cache. This is also configurable thru ENV VAR.
+        fetch_missing_chunks : bool (kwarg only)
+            If true, fetch any missing chunks needed to assemble the fileset.
+            If false, raise an error if any chunks are missing.
+        http_host_replacement : str
+            The value of this parameter replaces the host (including port) within
+            the presigned URLs when resolving CDRs. This configuration is useful
+            if, for some reason, the URL that pachd uses to interact with object
+            storage is different from the one that you use. For example, if your
+            Pachyderm object storage is running within your kubernetes cluster and
+            pachd is configured to use a URL which is only valid in cluster.
+            This may fix any issues, depending on the object storage and how they
+            generate presigned URLs.
+            Example: localhost:9000
+
+        Raises
+        ------
+            FileNotFoundError:
+                if chunk is missing from cache and fetch_missing_chunks=False.
+        """
+        destination = Path(os.path.expanduser(destination)).resolve()
+        destination.mkdir(parents=True, exist_ok=True)
+        resolver = CdrResolver(
+            cache_location=cache_location,
+            fetch_missing_chunks=fetch_missing_chunks,
+            http_host_replacement=http_host_replacement,
+        )
+
+        for msg in self.read_fileset_cdr(fileset_id=fileset, filters=[as_filter(path)]):
+            content = resolver.resolve(msg.ref)
+            file_path = msg.path
+            if file_path.startswith("/"):
```

Comment:
Why does this line need to happen? Is it worth writing a test to make sure that it does happen?

Reply:
Yeah I also paused when I wrote this line, and now that you've brought my attention to it, we don't need the branch at all. It can be replaced with:
```python
file_path = msg.path.removeprefix("/")
```
All PFS paths are absolute from the repo root. However, Python's pathlib implementation is a bit flawed (imo) in that if you try to join a path with an absolute path, the original path gets clobbered. For example `Path(/path/to/a/directory).joinpath("/file1")` evaluates to `Path(/file1)`. So we just need to convert the absolute PFS path into a relative path. 

I don't believe there's anything to test here, as `removeprefix` does nothing if the string doesn't have the prefix.  I'm going to remove the branch though and add a comment explaining this.

Reply:
Makes sense to me! agree nothing to test.